src/routes/auth.js

```diff
@@ -41,6 +41,18 @@ router.post("/register", async (req, res) => {
             "reason": "No username specified",
             "error_code": ERROR_TYPES.MISSING_FIELD,
         });
+    } else if (req.body.username.length > 20) {
+        return res.status(500).json({
+            "success": false,
+            "reason": "Username is too long",
+            "error_code": ERROR_TYPES.MISSING_FIELD,
+        });
+    } else if (req.body.username.length < 3) {
+        return res.status(500).json({
+            "success": false,
+            "reason": "Username is too short",
+            "error_code": ERROR_TYPES.MISSING_FIELD,
+        });
     }
```

Comment:
Would you oppose to moving this to a separate file (of validators)?

I'm thinking that this type of input checking could be done by middlewares (which are many times called validation middlewares or validators), and this would also reduce the friction with #38's changes (and goes along #19's ideas).

This would also reduce the code in the controller, making it simpler to understand :slightly_smiling_face: